server/src/lsp-server.spec.ts

```diff
@@ -645,6 +698,49 @@ describe('code actions', () => {
             }
         ]);
     }).timeout(10000);
+
+    it('provides organize imports for sub-actions of source.organizeImports', async () => {
+        server.didOpenTextDocument({
+            textDocument: doc
+        });
+        let result = (await server.codeAction({
+            textDocument: doc,
+            range: {
+                start: { line: 1, character: 29 },
+                end: { line: 1, character: 53 }
+            },
+            context: {
+                diagnostics: [{
+                    range: {
+                        start: { line: 1, character: 25 },
+                        end: { line: 1, character: 49 }
+                    },
+                    code: 6133,
+                    message: 'unused arg'
+                }],
+                only: ['source.organizeImports.test']
+            }
+        }))!;
+
+        // ensure this works on other peoples computers
+        try {
+            result = JSON.parse(JSON.stringify(result).replace(new RegExp(projectDir, 'g'), 'ROOT'));
+        } catch {
+            // this is ignored, since the matcher should fail if it fails, and the matcher will provide more useful output
+        }
+
+        assert.deepEqual(result, [
+            {
+                command: {
+                    arguments: ['ROOT/server/test-data/bar.ts'],
+                    command: '_typescript.organizeImports',
+                    title: ''
+                },
+                kind: 'source.organizeImports',
+                title: 'Organize imports'
+            }
+        ]);
```

Comment:
Requesting `source.organizeImports.test` should not result in code action being provided. The more specific `only` should not return the less specific `kind`.

Reply:
Got it - I've added more tests to more explicitly cover this.